activerecord/test/cases/transaction_callbacks_test.rb

```diff
@@ -83,6 +83,16 @@ def test_call_after_commit_after_transaction_commits
     assert_equal [:after_commit], @first.history
   end
 
+  def test_update_in_after_commit_on_create
+    new_record = TopicWithCallbacks.new(title: 'Topic')
+    new_record.after_commit_block(:create) do |r|
```

Comment:
I've saving a new record, it will be a create, not an update.

If this callback would have been called on update (inside after_commit) then it would lead to infinite recursion (like people mentioned in issue).

I've given you a wrong thought by using string ' updated' in code, it's an inappropriate label :smile: It was meant to show that I've updated record from inside the after_commit callback without recursion.